### DISCUSSION TITLE
Connecting different services and data sharing between them

### DISCUSSION THREAD

**Discussion by mathmul** [lencx/Noi#40] [Q&A] [answered]
Not sure of the title, but here's an explanation.

I often copy code from VSCode to ChatGPT for different reasons:

1. Something isn't working correctly and it help me identify the problem faster.
2. I have a similar code that I need already written and I want it to make appropriate changes.
3. I have a task that requires changes and/or additions of multiple files and I want it to help me basically from scratch.

While it would help with 1. and 2. as well, these two are pretty straightforward and quickly done with with the current approach. But with 3. I lose a lot of time explaining the architecture, and giving it codebase examples, and passing it all relevant models, controllers etc. and more often than not, by the time we get to the answer I seek, it seems to have already forgotten about many things it should already know.

So my question is: 

Can VSCode and ChatGPT inside Noi be used in a way so ChatGPT continuously has access to the codebase of the project opened in VSCode, so it would reduce the number of manually written prompts (albeit would likely use more queries to GPT API)?

**Comment by lencx** [comment, marked as answer]
You could try creating a custom GPT and then upload your code as a knowledge file into GPT, which might solve the issue you're facing.
